lib/action_client/applications/submission.rb

```diff
@@ -0,0 +1,18 @@
+module ActionClient
+  module Applications
+    class Submission
+      def initialize(app, after_action_callbacks)
+        @app = app
+        @after_action_callbacks = Array(after_action_callbacks).compact
+      end
+
+      def call(env)
+        response = @app.call(env)
+
+        @after_action_callbacks.reduce(response) do |(status, headers, body), callback|
```

Comment:
Maybe I'm missing a key tool, but as much as I tried, I couldn't figure out a way to re-use [ActiveSupport::Callbacks](https://api.rubyonrails.org/classes/ActiveSupport/Callbacks.html) or (more specifically) [AbstractController::Callbacks](https://api.rubyonrails.org/classes/AbstractController/Callbacks.html) in a way that used blocks accepted parameters.

The alternative to passing around the triplets is to create and store a `response` attribute, then read from and modify that in-line.

While that conceptually aligns with what a Controller _is_, I felt that the different in _direction_ was difficult to reason about, and had an uncanny valley effect:

Most Rails controllers' "response" is on the way out, whereas an `ActionClient` "response" is on the way in.